version/version.go

```diff
@@ -0,0 +1,75 @@
+package version
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// A Version is a version of the form <major>.<minor>.<patch>
+type Version struct {
+	Major int
+	Minor int
+	Patch int
+}
+
+// Make Builds a Version from major, minor, patch
+func Make(major int, minor int, patch int) Version {
+	return Version{major, minor, patch}
+}
+
+// FromString returns a Version based on a string
+func FromString(v string) (ver Version, err error) {
+	components := strings.Split(v, ".")
+	if len(components) != 3 {
+		return ver, fmt.Errorf("Version must contain 3 components: X.Y.Z")
+	}
+
+	maj, err := strconv.Atoi(components[0])
+	if err != nil {
+		return ver, fmt.Errorf("parsing %s as a version: %v", v, err)
+	}
+
+	min, err := strconv.Atoi(components[1])
+	if err != nil {
+		return ver, fmt.Errorf("parsing %s as a version: %v", v, err)
+	}
+
+	patch, err := strconv.Atoi(components[2])
```

Comment:
Correct. This is actually why I made the parsing so strict right now. It is \d.\d.\d and nothing else until we decide what else we want to support and how.